libs/@local/hash-backend-utils/src/service-usage.ts

```diff
@@ -145,18 +145,44 @@ export const createUsageRecord = async (
     userAccountId: AccountId;
   },
 ) => {
-  const properties: UsageRecord["properties"] = {
-    "https://hash.ai/@hash/types/property-type/input-unit-count/":
-      inputUnitCount,
-    "https://hash.ai/@hash/types/property-type/output-unit-count/":
-      outputUnitCount,
+  const properties: UsageRecord["propertiesWithMetadata"] = {
+    value: {
+      ...(inputUnitCount
+        ? {
+            "https://hash.ai/@hash/types/property-type/input-unit-count/": {
+              value: inputUnitCount,
+              metadata: {
+                dataTypeId:
+                  "https://blockprotocol.org/@blockprotocol/types/data-type/number/v/1",
+              },
+            },
+          }
+        : {}),
+      ...(outputUnitCount
+        ? {
```

Comment:
Good point. Is there a way to lint against this behavior? Rust for example does not support implicit conversions because of exactly this reason.
I think I changed all locations.

Reply:
Hopefully I now have all usages. (A working search in JetBrains would help)

Reply:
We can probably achieve this using a custom rule – I created H-3109 for it